### PR TITLE
Schema-4.5 cleanup. (DOI model)

### PR DESCRIPTION
**Pull request by svogt0511**
## Purpose

Related to: Issue #1016, [PR#1112](https://github.com/datacite/lupo/pull/1112)

## Approach

STEP 1 - This PR which obsoletes the publisher column from the database by making it an active record virtual attribute.  To preserve backward compatibility, 'publisher' is still accepted by the API as either a string or a hash, but is saved to the DB or read from the DB as publisher_obj.  Also, the API publisher='true' flag still works as documented to return either a string or hash in the response.

STEP 2 - A second PR ([PR#1112](https://github.com/datacite/lupo/pull/1112)) which has the rake task to remove the publisher column from the DB, and contains the new schema.rb file.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)

- [ ] New feature (non-breaking change which adds functionality)

- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Reviewer, please remember our [guidelines](https://datacite.atlassian.net/wiki/spaces/TEC/pages/1168375809/Pull+Request+Guidelines):

- Be humble in the language and feedback you give, ask don't tell.
- Consider using positive language as opposed to neutral when offering feedback. This is to avoid the negative bias that can occur with neutral language appearing negative.
- Offer suggestions on how to improve code e.g. simplification or expanding clarity.
- Ensure you give reasons for the changes you are proposing.

### CODE REVIEW

File: app/models/doi.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 class Doi < ApplicationRecord
   PUBLISHER_JSON_SCHEMA = "#{Rails.root}/app/models/schemas/doi/publisher.json"
-  audited only: %i[doi url creators contributors titles publisher publisher_obj publication_year types descriptions container sizes formats version_info language dates identifiers related_identifiers related_items funding_references geo_locations rights_list subjects schema_version content_url landing_page aasm_state source reason]
```
Comment (digitaldogsbody) [inline app/models/doi.rb:8]: Will removing this remove historical data about changes to the field from the `activities` table/index/endpoint?

**Review by digitaldogsbody — APPROVED** [review]
Looks good - one question about the impact of the audit change, but it's non-blocking to merge

File: app/models/doi.rb (hunk 1 above)
Comment (svogt0511) [reply app/models/doi.rb:8]: Great question.  Deleting the field name from that line just tells the audit module (activities) to not record changes for that field.  It won't erase historical data for publisher or leave those changes out of a call to /activities. 

Changes are kept in the audited_changes column of the audits table.  For reporting changes, activities just grabs audited_changes, but doesn't care what is in it. For recording changes, the publisher column will be gone.

**Review by ashwinisukale — APPROVED** [review]
Looks Good

**Pull request merged**